Source/WebGPU/WebGPU/Device.mm

```diff
@@ -487,6 +487,313 @@ static void setOwnerWithIdentity(id<MTLResourceSPI> resource, auto webProcessID)
+    }
+
+    MTLRenderPipelineDescriptor* mtlRenderPipelineDescriptor = [MTLRenderPipelineDescriptor new];
+    mtlRenderPipelineDescriptor.vertexFunction = isUint16 ? functionUshort : function;
+    mtlRenderPipelineDescriptor.rasterizationEnabled = false;
+    mtlRenderPipelineDescriptor.rasterSampleCount = rasterSampleCount;
+    mtlRenderPipelineDescriptor.fragmentFunction = nil;
+    mtlRenderPipelineDescriptor.inputPrimitiveTopology = MTLPrimitiveTopologyClassPoint;
+
+    if (isUint16) {
+        if (rasterSampleCount > 1)
+            result = m_indexBufferClampUshortPSOMS = [m_device newRenderPipelineStateWithDescriptor:mtlRenderPipelineDescriptor error:&error];
+        else
+            result = m_indexBufferClampUshortPSO = [m_device newRenderPipelineStateWithDescriptor:mtlRenderPipelineDescriptor error:&error];
+    } else {
+        if (rasterSampleCount > 1)
+            result = m_indexBufferClampUintPSOMS = [m_device newRenderPipelineStateWithDescriptor:mtlRenderPipelineDescriptor error:&error];
+        else
+            result = m_indexBufferClampUintPSO = [m_device newRenderPipelineStateWithDescriptor:mtlRenderPipelineDescriptor error:&error];
+    }
+
+    if (error) {
+        WTFLogAlways("%@", error);
+        return nil;
+    }
+    return result;
+}
+
+id<MTLRenderPipelineState> Device::indexedIndirectBufferClampPipeline(NSUInteger rasterSampleCount)
+{
+    if (!m_device)
+        return nil;
+
+    id<MTLRenderPipelineState> result = rasterSampleCount > 1 ? m_indexedIndirectBufferClampPSOMS : m_indexedIndirectBufferClampPSO;
+    if (result)
+        return result;
+
+    static id<MTLFunction> function = nil;
+    NSError *error = nil;
+    if (!function) {
+        MTLCompileOptions* options = [MTLCompileOptions new];
+        ALLOW_DEPRECATED_DECLARATIONS_BEGIN
+        options.fastMathEnabled = YES;
+        ALLOW_DEPRECATED_DECLARATIONS_END
+        /* NOLINT */ id<MTLLibrary> library = [m_device newLibraryWithSource:@R"(
+    using namespace metal;
+    [[vertex]] void vs(device const MTLDrawIndexedPrimitivesIndirectArguments& input [[buffer(0)]], device MTLDrawIndexedPrimitivesIndirectArguments& indexedOutput [[buffer(1)]], device MTLDrawPrimitivesIndirectArguments& output [[buffer(2)]], const constant uint* indexBufferCount [[buffer(3)]]) {
+        indexedOutput.indexCount = metal::select(input.indexCount, 0u, indexBufferCount[0] > input.indexCount + input.indexStart);
+        indexedOutput.instanceCount = input.instanceCount;
+        indexedOutput.indexStart = metal::select(input.indexStart, 0u, indexBufferCount[0] > input.indexCount + input.indexStart);
+        indexedOutput.baseVertex = input.baseVertex;
+        indexedOutput.baseInstance = input.baseInstance;
+
+        output.vertexCount = metal::select(input.indexCount, indexBufferCount[0], indexBufferCount[0] > input.indexCount);
+        output.instanceCount = 1;
+        output.vertexStart = input.baseVertex;
```

Comment:
this should be:
```
output.vertexStart = input.indexStart;
```

as the goal is to run over the index buffer in the subsequent draw call